src/main/scala/is/hail/linalg/BlockMatrix.scala

```diff
@@ -149,68 +149,29 @@ object BlockMatrix {
   object ops {
 
     implicit class Shim(l: M) {
-      def t: M =
-        l.transpose()
-
-      def *(r: M): M =
-        l.multiply(r)
-
-      def *(r: BDM[Double]): M =
-        l.multiply(r)
-
-      def +:+(r: M): M =
-        l.add(r)
-
-      def -:-(r: M): M =
-        l.subtract(r)
-
-      def *:*(r: M): M =
-        l.pointwiseMultiply(r)
-
-      def /:/(r: M): M =
-        l.pointwiseDivide(r)
-
-      def +(r: Double): M =
-        l.scalarAdd(r)
-
-      def -(r: Double): M =
-        l.scalarSubtract(r)
-
-      def *(r: Double): M =
-        l.scalarMultiply(r)
-
-      def /(r: Double): M =
-        l.scalarDivide(r)
-
-      def :+(v: Array[Double]): M =
-        l.vectorAddToEveryColumn(v)
-
-      def :*(v: Array[Double]): M =
-        l.vectorPointwiseMultiplyEveryColumn(v)
-
-      def --+(v: Array[Double]): M =
-        l.vectorAddToEveryRow(v)
-
-      def --*(v: Array[Double]): M =
-        l.vectorPointwiseMultiplyEveryRow(v)
+      def +(r: M): M = l.add(r)
+      def -(r: M): M = l.sub(r)
+      def *(r: M): M = l.mul(r)
+      def /(r: M): M = l.div(r)
+      
+      def +(r: Double): M = l.scalarAdd(r)
+      def -(r: Double): M = l.scalarSub(r)
+      def *(r: Double): M = l.scalarMul(r)
+      def /(r: Double): M = l.scalarDiv(r)
+
+      def %*%(r: M): M = l.matrixMultiply(r)
+      def %*%(r: BDM[Double]): M = l.matrixMultiply(r)
```

Comment:
I'm on record as being against this.

Reply:
I guess I'm accustomed from R, same precedence as `*`. But yeah, I think it's best to make it `dot` again (I just didn't want `matrixMultiply`).